src/renderer/hooks/useInstantOnboarding.ts

```diff
@@ -0,0 +1,196 @@
+import { useCallback, useContext } from 'react'
+
+import useConfirmationDialog from './useConfirmationDialog'
+import useDialog from '../hooks/useDialog'
+import useSecureJoin from './useSecureJoin'
+import useTranslationFunction from './useTranslationFunction'
+import { BackendRemote } from '../backend-com'
+import { ConfigureProgressDialog } from '../components/LoginForm'
+import { InstantOnboardingContext } from '../contexts/InstantOnboardingContext'
+import { ScreenContext } from '../contexts/ScreenContext'
+import { Screens } from '../ScreenController'
+
+import type { QrWithUrl } from './useProcessQr'
+
+const DEFAULT_CHATMAIL_INSTANCE_URL =
+  'https://nine.testrun.org/cgi-bin/newemail.py'
+
+export default function useInstantOnboarding() {
+  const context = useContext(InstantOnboardingContext)
+  const openConfirmationDialog = useConfirmationDialog()
+  const tx = useTranslationFunction()
+  const { openDialog } = useDialog()
+  const { screen, changeScreen } = useContext(ScreenContext)
+  const { secureJoinContact, secureJoinGroup } = useSecureJoin()
+
+  if (!context) {
+    throw new Error(
+      'useInstantOnboarding has to be used within <InstantOnboardingProvider>'
+    )
+  }
+
+  const {
+    setShowInstantOnboarding,
+    setWelcomeQr,
+    showInstantOnboarding,
+    welcomeQr,
+  } = context
+
+  const switchToInstantOnboarding = useCallback(
+    async (qrWithUrl?: QrWithUrl) => {
+      if (qrWithUrl) {
+        const { qr } = qrWithUrl
+        if (
+          !['account', 'askVerifyGroup', 'askVerifyContact'].includes(qr.kind)
+        ) {
+          throw new Error(
+            'QR code needs to be of kind `account`, `askVerifyGroup` or `askVerifyContact` for instant onboarding flow'
+          )
+        }
+
+        // Ask user to confirm creating a new account if they already have one
+        const numberOfAccounts = (await BackendRemote.rpc.getAllAccountIds())
+          .length
+        if (qr.kind === 'account' && numberOfAccounts > 0) {
+          const message: string =
+            numberOfAccounts === 1
+              ? 'qraccount_ask_create_and_login'
+              : 'qraccount_ask_create_and_login_another'
+          const replacementValue = qr.domain
+          const userConfirmed = await openConfirmationDialog({
+            message: tx(message, replacementValue),
+            confirmLabel: tx('login_title'),
+          })
+
+          if (!userConfirmed) {
+            return
+          }
+        }
+      }
+
+      setWelcomeQr(qrWithUrl)
+      setShowInstantOnboarding(true)
+
+      setTimeout(async () => {
+        if (screen !== Screens.Welcome) {
+          // Log out first by switching to an (temporary) blank account
+          const blankAccount = await BackendRemote.rpc.addAccount()
+          window.__selectAccount(blankAccount)
+          changeScreen(Screens.Welcome)
```

Comment:
selecting an unconfigured account should bring you always into the welcome screen, so no need to change to it again?

Reply:
Right, I've added it anyhow as react de-deduplicates it and on top it makes the code easier to read, but I can remove it if it's confusing

Reply:
rather remove it and make a comment that says that it automatically goes there, maybe with a link to the file where it continues